### PR TITLE
Align submission of params with DSA openapi

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
DSC was submitting some params with methods that were not consistent with DSA openapi. The new version of committee will be enforcing this more rigorously than the previous.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit (will be running integration test after cutting new gem)

### CODE REVIEW

File: lib/dor/services/client/object_version.rb [hunk 1]
```diff
@@ -79,7 +78,7 @@ def open(**params)
         # @return [String] a message confirming successful closing
         def close(**params)
           resp = connection.post do |req|
-            req.url close_version_path
+            req.url with_querystring(url: close_version_path, params: params)
             req.headers['Content-Type'] = 'application/json'
             req.body = params.to_json if params.any?
```
Comment (jmartin-sul) [inline lib/dor/services/client/object_version.rb:83]: does the `req.body = ...` need to be dropped here like it was above?

File: lib/dor/services/client/objects.rb [hunk 2]
```diff
@@ -14,11 +14,10 @@ class Objects < VersionedService
         # @return [Cocina::Models::DROWithMetadata,Cocina::Models::CollectionWithMetadata,Cocina::Models::AdminPolicyWithMetadata] the returned model
         def register(params:, assign_doi: false, validate: false)
           resp = connection.post do |req|
-            req.url objects_path
+            req.url with_querystring(url: objects_path, params: { assign_doi: assign_doi })
             req.headers['Content-Type'] = 'application/json'
             # asking the service to return JSON (else it'll be plain text)
             req.headers['Accept'] = 'application/json'
-            req.params[:assign_doi] = true if assign_doi
             req.body = params.to_json
```
Comment (jmartin-sul) [inline lib/dor/services/client/objects.rb:21]: similar question here about leaving `req.body = ...`

**Review by jmartin-sul — COMMENTED** [review]
a question about consistency of `req.body = ...` cleanup, but otherwise LGTM

File: lib/dor/services/client/object_version.rb (hunk 1 above)
Comment (justinlittman) [reply lib/dor/services/client/object_version.rb:83]: Yes. Fixed.

File: lib/dor/services/client/objects.rb (hunk 2 above)
Comment (justinlittman) [reply lib/dor/services/client/objects.rb:21]: No, these are different params (but unfortunately have the same argument name, so confusing).

**Review by jmartin-sul — APPROVED** [review]

**Pull request merged**